contracts/mainnet/connectors/wsteth/main.sol

```diff
@@ -0,0 +1,62 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+import { Basic } from "../../common/basic.sol";
+import './helpers.sol';
+
+/**
+ * @title WSTETH.
+ * @dev Wrap and Unwrap STETH.
+*/
+
+abstract contract WSTETHContract is Helpers, Basic {
+
+    /**
+     * @dev Deposit STETH into WSTETH.
+     * @notice Wrap STETH into WSTETH
+     * @param stethAmt The amount of STETH to deposit. (For max: `uint256(-1)`)
+     * @param getId ID to retrieve amt.
+     * @param setId ID stores the amount of STETH deposited.
+     */
+    function deposit(
+        uint256 stethAmt,
+        uint256 getId,
+        uint256 setId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+
+        uint256 _amt = getUint(getId, stethAmt);
+        _amt = _amt == uint(-1) ? _amt = stethContract.balanceOf(address(this)) : _amt;
+
+        approve(stethContract, address(wstethContract), _amt);
+
+        setId = wstethContract.wrap(_amt);
```

Comment:
@shriyatyagii we need to set the amount in setId and use setUnit amount.

Reply:
Done

Reply:
Done